src/etc/inc/util.inc

```diff
@@ -1932,19 +1932,30 @@ function mac_format($clientmac) {
 	}
 }
 
-function resolve_retry($hostname, $retries = 5) {
-
-	if (is_ipaddr($hostname)) {
-		return $hostname;
-	}
+function resolve_retry($hostname, $protocol = 'any', $retries = 5) {
 
 	for ($i = 0; $i < $retries; $i++) {
-		// FIXME: gethostbyname does not work for AAAA hostnames, boo, hiss
-		$ip = gethostbyname($hostname);
 
-		if ($ip && $ip != $hostname) {
-			/* success */
-			return $ip;
+		// IPv4 or any, return IPv4 address if resolved
+		if ($protocol != 'inet6') {
+			if (is_ipaddrv4($hostname)) {
+                                return $hostname;
+                        }
+                        $ip = dns_get_record($hostname, DNS_A)[0]['ip'];
```

Comment:
This should probably be more than one step here, since if it fails the return false will be `false` and then this attempt would fail with a PHP error since the response would not be an array.

Reply:
I found that I got "Warning: dns_get_record(): DNS Query failed " error in any case, 
if I add 
php```
if (dns_get_record($hostname, $dnsp)) {
 ....
} else {
continue
} 
```
I got "Warning: dns_get_record(): DNS Query failed " error anyway